src/tools/diagnostic-messages.md

```diff
@@ -11836,6 +11836,39 @@ void f(int x) {}
 void g({required int x}) {}
 {% endprettify %}
 
+### missing_dependency
+
+_Missing a dependency on imported package '{0}'._
+
+#### Description
+
+The analyzer produces this diagnostic when there's a package that has been
+imported in the source but is not listed in the dependency of the
+importing package.
+
+#### Example
+
+The following code produces this diagnostic because the package `path` is
+not listed in the dependencies, while there is an import statement
+with package `path` in the source code of package example:
+
+{% prettify yaml tag=pre+code %}
+name: example
+dependencies:
+  meta: ^1.0.2
+{% endprettify %}
```

Comment:
**question:** Why does this use `prettify` instead of the Markdown code block?

Reply:
Until I finish standardizing the highlighting support, all code blocks in the diagnostic messages are generated with prettify. 

Once I finish the standardization work, I'll update the generator to generate markdown code blocks instead.